src/api/usersRouter.ts

```diff
@@ -62,15 +67,43 @@ usersRouter.post('/:userId/resendInvitation', async (req: UserRequest, res) => {
       return res.sendStatus(404);
     }
     if (resultLength > 1) {
-        errorLogger.error(
-          `Multiple results received when loading user entry for ${req.user?.email}`
-        );
+      errorLogger.error(`Multiple results received when loading user entry for ${req.user?.email}`);
       return res.sendStatus(500);
     }
 
     logger.info(`Resending invitation email for ${req.user?.email}, keycloak ID ${user[0].id}`);
     await sendInviteEmail(kcAdminClient, user[0]);
     return res.sendStatus(200);
-});
+  });
+
+  usersRouter.delete('/:userId', async (req: UserRequest, res) => {
+    const { user } = req;
+    if (req.auth?.payload?.email === user?.email) {
+      return res.status(403).send([{ message: 'You do not have permission to delete yourself.' }]);
+    }
+    const kcAdminClient = await getKcAdminClient();
+    await deleteUserByEmail(kcAdminClient!, user?.email!);
+    await user!.$query().delete();
+    return res.sendStatus(200);
+  });
+
+  const UpdateUserParser = z.object({
+    firstName: z.string(),
+    lastName: z.string(),
+    role: z.nativeEnum(UserRole),
+  });
+
+  usersRouter.patch('/:userId', async (req: UserRequest, res) => {
+    const { user } = req;
+    const data = UpdateUserParser.parse(req.body);
+    const kcAdminClient = await getKcAdminClient();
```

Comment:
Minor (doesn't need to be fixed unless you want to!) - the database update and the keycloak update are independent, so you don't need to wait for one to finish before starting the other - you could start them both and then await them both.

The same applies to the delete.